pkg/common/cns-lib/volume/listview.go

```diff
@@ -0,0 +1,360 @@
+package volume
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/property"
+	"github.com/vmware/govmomi/session"
+	"github.com/vmware/govmomi/view"
+	"github.com/vmware/govmomi/vim25/types"
+
+	cnsvsphere "sigs.k8s.io/vsphere-csi-driver/v2/pkg/common/cns-lib/vsphere"
+	"sigs.k8s.io/vsphere-csi-driver/v2/pkg/csi/service/logger"
+)
+
+const (
+	// in case of vc connection failure, we wait for a minute
+	// to not overwhelm the vc with continuous requests
+	waitForUpdatesRetry = 1 * time.Minute
+	// `info` is a defined property in a Task object
+	// we want to monitor the TaskInfo that includes the status of a task
+	infoPropertyName = "info"
+)
+
+// ListViewImpl is the struct used to manage a single listView instance.
+type ListViewImpl struct {
+	// sync.RWMutex: a mutex is required for concurrent access to a go map
+	// reference - see the section on Concurrency at https://go.dev/blog/maps
+	// important note - sync.RWMutex can be held by an arbitrary number of readers or a single writer.
+	// here's a short example of how sync.RWMutex works - https://go.dev/play/p/Gm22qn-lpq-
```

Comment:
The go.dev/play link will expire, don't post it.

Reply:
TIL. I've removed it